raster-foundry/granary
# Add simple token auth

## Overview
Add basic token auth to the prediction endpoints

### Checklist

- [x] Any new migrations have been audited to make sure they won't kill the application while being applied
- [ ] ~~Any new API endpoints have tests~~ No way to add tokens by the API, so tests have auth disabled for now
- [ ] ~~Any frontend changes are hidden behind a feature flag~~

### Demo
```
> $ http --auth-type=jwt localhost:8080/api/predictions 
HTTP/1.1 200 OK
Content-Length: 2
Content-Type: application/json
Date: Thu, 19 Dec 2019 15:32:34 GMT

[]
```

```
> $ http localhost:8080/api/predictions
HTTP/1.1 403 Forbidden
Content-Length: 24
Content-Type: application/json
Date: Thu, 19 Dec 2019 15:33:26 GMT

{
    "msg": "Not authorized"
}
```

### TODO
- [x] model endpoints auth

## Testing Instructions

- build the jar and run migrations
- start the server
- make a request to a predictions endpoint `http :8080/api/predictions` and verify that a 403 is returned
- Add a token to the database and use it to make another request, verify that it completes with a 200
```
export JWT_AUTH_TOKEN="{token}"
http --auth-type=jwt :8080/api/predictions
```

Closes #47
